scalr/resource_scalr_service_account_token.go

```diff
@@ -0,0 +1,146 @@
+package scalr
+
+import (
+	"context"
+	"errors"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scalr/go-scalr"
+)
+
+func resourceScalrServiceAccountToken() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalrServiceAccountTokenCreate,
+		ReadContext:   resourceScalrServiceAccountTokenRead,
+		UpdateContext: resourceScalrServiceAccountTokenUpdate,
+		DeleteContext: resourceScalrServiceAccountTokenDelete,
+		Schema: map[string]*schema.Schema{
+			"service_account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"token": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceScalrServiceAccountTokenCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+
+	saID := d.Get("service_account_id").(string)
+
+	options := scalr.AccessTokenCreateOptions{}
+	if desc, ok := d.GetOk("description"); ok {
+		options.Description = scalr.String(desc.(string))
+	}
+
+	log.Printf("[DEBUG] Create access token for service account: %s", saID)
+	at, err := scalrClient.ServiceAccountTokens.Create(ctx, saID, options)
+	if err != nil {
+		return diag.Errorf(
+			"Error creating access token for service account %s: %v", saID, err)
+	}
+
+	// the token is returned from API only while creating
+	_ = d.Set("token", at.Token)
+
+	d.SetId(at.ID)
+
+	return resourceScalrServiceAccountTokenRead(ctx, d, meta)
+}
+
+func resourceScalrServiceAccountTokenRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+	id := d.Id()
+	saID := d.Get("service_account_id").(string)
+
+	if saID == "" {
+		return diag.Errorf("This resource does not support import")
+	}
+
+	log.Printf("[DEBUG] Read service account token: %s", id)
+	options := scalr.AccessTokenListOptions{}
+
+	for {
+		atl, err := scalrClient.ServiceAccountTokens.List(ctx, saID, options)
+
+		if err != nil {
+			if errors.Is(err, scalr.ErrResourceNotFound) {
+				log.Printf("[DEBUG] service account %s not found", saID)
+				d.SetId("")
+				return nil
+			}
+			return diag.Errorf("Error reading service account token %s: %v", id, err)
+		}
+
+		for _, at := range atl.Items {
+			if at.ID == id {
```

Comment:
@petroprotsakh if we only need one access token by ID why don't we just use `/service-accounts/<service_account_model:service_account>` endpoint? why it is not published?

Reply:
@soltysss there is no api endpoint for getting single service account's token. `/service-accounts/<service_account_model:service_account>` returns service account, not token.